boards/arm/stm32l476g_disco/stm32l476g_disco.yaml

```diff
@@ -0,0 +1,7 @@
+identifier: stm32l476g_disco
+name: 32L476GDISCOVERY
```

Comment:
Missing "STM" in front of this, maybe?

Reply:
I believe the name of the board is correct. See: http://www.st.com/en/evaluation-tools/32l476gdiscovery.html
I've copied the template on the STM32L476G Discovery board.

Reply:
I don't think so, but perhaps @erwango can clear this up:

```
$ grep name: boards/arm/st*/*.yaml
stm3210c_eval/stm3210c_eval.yaml:name: STM3210C-EVAL
stm32373c_eval/stm32373c_eval.yaml:name: STM32373C-EVAL
stm32f3_disco/stm32f3_disco.yaml:name: STM32F3DISCOVERY
stm32f412g_disco/stm32f412g_disco.yaml:name: 32F412GDISCOVERY
stm32f429i_disc1/stm32f429i_disc1.yaml:name: STM32F429I_DISC1
stm32f469i_disco/stm32f469i_disco.yaml:name: 32F469IDISCOVERY
stm32f4_disco/stm32f4_disco.yaml:name: STM32F4DISCOVERY
stm32l496g_disco/stm32l496g_disco.yaml:name: 32L496GDISCOVERY
stm32_min_dev/stm32_min_dev.yaml:name: STM32-MIN-DEV
stm32_mini_a15/stm32_mini_a15.yaml:name: STM32 Mini A15
```

Reply:
Do we know the purpose of this field?
Problem is ST board naming convention is not uniform across boards and neither across media. Most simple and most consistent rule I can find is to use silkscreen naming: STM32L496G-DISCO.
This would be same (to some extent) for XX-EVAL boards.
We keep NUCLEO boards as NUCLEO-XX
Exception to this rule (to make it a proven rule) would be disco iot board : B-L475E-IOT1 (Don't ask me why)

Finally STM32-MIN... are not ST boards.